src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionResolveHandler.cs

```diff
@@ -0,0 +1,204 @@
+            // additional suggestion a code action can make about what to do post edit.
+            var applicableActions = operations.Where(o => o is ApplyChangesOperation or DocumentNavigationOperation).ToArray();
+
+            // TO-DO: We currently must execute code actions which add new documents on the server as commands,
+            // since there is no LSP support for adding documents yet. In the future, we should move these actions
+            // to execute on the client.
+            // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/
+
+            var solution = document.Project.Solution;
+            var textDiffService = solution.Services.GetService<IDocumentTextDifferencingService>();
+
+            using var _ = ArrayBuilder<TextDocumentEdit>.GetInstance(out var textDocumentEdits);
+
+            foreach (var option in operations)
+            {
+                // We only support making solution-updating operations in LSP.  And only ones that modify documents. 1st
+                // class code actions that do more than this are supposed to add the CodeAction.MakesNonDocumentChange
+                // in their Tags so we can filter them out before returning them to the client.
+                //
+                // However, we cannot enforce this as 3rd party fixers can still run.  So we filter their results to 
+                // only apply the portions of their work that updates documents, and nothing else.
+                if (option is not ApplyChangesOperation applyChangesOperation)
+                    continue;
+
+                var changes = applyChangesOperation.ChangedSolution.GetChanges(solution);
+                var projectChanges = changes.GetProjectChanges();
+
+                // ignore any non-document changes.  In the future we could see if there is some mechanism by which we
+                // can make these changes safely in LSP.
+#if false
+
+                // TO-DO: If the change involves adding or removing a document, execute via command instead of WorkspaceEdit
+                // until adding/removing documents is supported in LSP: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/
+                // After support is added, remove the below if-statement and add code to support adding/removing documents.
+                var addedDocuments = projectChanges.SelectMany(
+                    pc => pc.GetAddedDocuments().Concat(pc.GetAddedAdditionalDocuments().Concat(pc.GetAddedAnalyzerConfigDocuments())));
+                var removedDocuments = projectChanges.SelectMany(
+                    pc => pc.GetRemovedDocuments().Concat(pc.GetRemovedAdditionalDocuments().Concat(pc.GetRemovedAnalyzerConfigDocuments())));
+                if (addedDocuments.Any() || removedDocuments.Any())
+                {
+                    codeAction.Command = SetCommand(codeAction.Title, data);
```

Comment:
since we removed the RunCodeActionHandler we should likely also remove all places where we set the command (since theres nothing to run the command any longer)

Reply:
nvm its in the #if false.  we can look at this later.